File: assignment2/part2/section3/memory_demo.cpp
#include <iostream>

using namespace std;

int main() {
    int* numbers = new int[1000000];
    cout << "Allocated 1000000 integers" << endl;

    numbers[0] = 42;
    cout << "First value: " << numbers[0] << endl;

    delete[] numbers;
    numbers = nullptr;

    cout << "Memory released manually with delete[]" << endl;
    return 0;
}

File: assignment2/part2/section3/memory_demo.rs
fn main() {
    let numbers = vec![0_u64; 1_000_000];
    println!("Allocated {} integers", numbers.len());

    let first = &numbers[0];
    println!("Borrowed first value: {}", first);

    consume(numbers);
}

fn consume(data: Vec<u64>) {
    println!("Ownership moved into consume(): {}", data.len());
}
